File: server/Models/Notebook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace dungeon_noter.Models;

public class Notebook : RepoItem
{
  public string Title { get; set; }
  public string Icon { get; set; }
  public string Color { get; set; } = "#ffffff";
  public string CoverImg { get; set; }
  public int EntryCount { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }

}

File: client/src/models/Notebook.js
export class Notebook{
  constructor(data){
    this.id = data.id
    this.title = data.title
    this.coverImg = data.coverImg
    this.icon = data.icon
    this.color = data.color
    this.entryCount = data.entryCount
    this.creatorId = data.creatorId
    this.creator = data.creator
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
  }

  get updatedTimeStamp(){
    return this.updatedAt.toLocaleString('en-US', {weekday: 'long', month:'short', year: 'numeric', day: 'numeric', hour: '2-digit', minute: '2-digit'}).replace(/\//g, '.')
  }
  get createdTimeStamp(){
    return this.createdAt.toLocaleString('en-US', {weekday: 'long', month:'short', year: 'numeric', day: 'numeric'}).replace(/\//g, '.')
  }
}
